dags/synapse-dataset-to-croissant.py

```diff
@@ -381,37 +356,55 @@ def construct_distribution_section_for_files(files_attached_to_dataset: List[Fil
         SELECT
             vd.id,
             vd.versionNumber,
-            vd.contentMd5
+            vd.contentMd5,
+            vd.fileHandleId,
+            COALESCE(fl.content_type, 'NOT_SET') as CONTENT_TYPE,
         FROM version_data AS vd
+        LEFT JOIN synapse_data_warehouse.synapse.file_latest fl
+            ON fl.id = vd.fileHandleId
         WHERE (vd.id, vd.versionNumber) IN ({', '.join(['(%s, %s)'] * (len(id_and_version_pairs)//2))});
         """
         try:
             cs.execute(
                 query,
                 (ids_of_files + id_and_version_pairs),
             )
-            file_md5s = cs.fetch_pandas_all()
+            file_md5_and_types = cs.fetch_pandas_all()
         finally:
             cs.close()
 
     for file in files_attached_to_dataset:
         file: File = file
-        if file.file_handle and file.file_handle.content_md5:
-            file_md5 = file.file_handle.content_md5
-        elif not file_md5s.empty and int(file.id.replace("syn", "")) in file_md5s["ID"].values:
-            file_md5 = file_md5s.loc[file_md5s["ID"] == int(
+        if not file_md5_and_types.empty and int(file.id.replace("syn", "")) in file_md5_and_types["ID"].values:
+            file_md5 = file_md5_and_types.loc[file_md5_and_types["ID"] == int(
                 file.id.replace("syn", "")), "CONTENTMD5"].values[0]
+            
+            file_content_type = file_md5_and_types.loc[file_md5_and_types["ID"] == int(
+                file.id.replace("syn", "")), "CONTENT_TYPE"].values[0]
         else:
             file_md5 = "unknown_md5"
 
+        if not file_content_type or file_content_type == "NOT_SET":
+            if file.file_handle and file.file_handle.file_name:
+                file_content_type = mimetypes.guess_type(
+                    file.file_handle.file_name, strict=False)[0]
+            else:
+                file_content_type = mimetypes.guess_type(
+                    file.name, strict=False)[0]
+                
+            if not file_content_type:
+                # For binary documents without a specific or known subtype, application/octet-stream should be used.
+                # Source: https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/MIME_types
+                file_content_type = "application/octet-stream"
+
         distribution_files.append(
             {
                 "@type": "FileObject",
                 "@id": f"{file.id}.{file.version_number}",
                 "name": f"{file.name}",
-                "description": f"Data file associated with {file.name}",
+                "description": file.description if file.description else f"Data file associated with {file.name}",
```

Comment:
can file.name be None?

Reply:
No:

```

SELECT * FROM 
    SYNAPSE_DATA_WAREHOUSE.SYNAPSE.NODE_LATEST
WHERE
    node_type = 'file'
    AND NAME is null
limit 1

SELECT * FROM 
    SYNAPSE_DATA_WAREHOUSE.SYNAPSE.NODE_LATEST
WHERE
    node_type = 'file'
    AND NAME = ''
limit 1
```


Both return no results from the datawarehouse